src/compiler/checker.ts

```diff
@@ -60,6 +72,35 @@ namespace ts {
         undefinedSymbol.declarations = [];
         const argumentsSymbol = createSymbol(SymbolFlags.Property | SymbolFlags.Transient, "arguments");
 
+        const {typeToString, getSymbolDisplayBuilder, symbolToString, signatureToString, typePredicateToString, visibilityToString} =
+            createWriter(
+                createSymbol,
+                getSymbolOfNode,
+                compilerOptions,
+                languageVersion,
+                getTypeParametersOfClassOrInterface,
+                getAccessibleSymbolChain,
+                needsQualification,
+                getQualifiedLeftMeaning,
+                hasExternalModuleSymbol,
+                isTypeAny,
+                () => nothingType,

```

Comment:
Since you only call this once, it might be beneficial to pass an arguments bag - especially for something like `() => nothingType` where it's not obvious what this is used for